index.js

```diff
@@ -0,0 +1,76 @@
+const express = require('express');
+const bcrypt = require('bcryptjs');
+
+const db = require('./data/helpers');
+
+const server = express();
+
+server.use(express.json());
+
+// REGISTER
+server.post('/api/register', async (req, res) => {
+  // get form input values
+  const registrationData = req.body;
+  // validate data
+  if (!registrationData.username || !registrationData.password) {
+    return res
+      .status(400)
+      .json({ message: 'Please enter a valid username and password.' });
+  }
+  // Check if user already exists
+  try {
+    const userInDb = db.getByUsername(registrationData.username);
```

Comment:
you need to `await` this if you want it to stop the rest of your code from running if the username already exists